packages/rum-core/src/domain/profiling/exporter/exportToJsonIntake.ts

```diff
@@ -0,0 +1,165 @@
+import type { Payload} from 'packages/core/src/transport/httpRequest';
+import { sendXHR } from 'packages/core/src/transport/httpRequest';
+import type { EndpointBuilder, InitConfiguration } from '@datadog/browser-core';
+import type {
+    RumProfilerTraceExporter,
+    RumProfilerTrace,
+} from '../types';
+import { getLongTaskId } from '../utils/longTaskRegistry';
+
+import { getEmptyPprofBlob } from './emptyPprof';
+
+interface ProfileEventAttributes {
+    application: { id: string };
+    session?: { id: string };
+    view?: { name: string[] };
+    context?: { profile_long_task_id: string[] };
+}
+interface ProfileEvent extends ProfileEventAttributes{
+    attachments: string[];
+    start: string // ISO date
+    end:  string // ISO date
+    family: 'chrome';
+    tags_profiler: string;
+}
+
+/**
+ * Exports RUM profile as JSON to public profiling intake.
+ */
+export const exportToJSONIntake: RumProfilerTraceExporter = (
+    profilerTrace,
+    endpointBuilder,
+    applicationId,
+    sessionId,
+) => {
+    const event = createProfileEvent(profilerTrace, endpointBuilder, applicationId, sessionId);
+    const payload = createProfilePayload(profilerTrace, event)
+
+    // Sends profile to public profiling intake.
+    const xhrUrl = endpointBuilder.build('xhr', payload)
+
+    // Send everything.
+    return new Promise((resolve) =>  sendXHR(xhrUrl, payload.data, resolve));
+};
+
+function createProfileEvent(
+    profilerTrace: RumProfilerTrace, 
+    endpointBuilder: EndpointBuilder, 
+    applicationId: string ,
+    sessionId: string | undefined
+): ProfileEvent {
+    const rawParameters = endpointBuilder.rawParameters;
```

Comment:
As commented previously, this is probably not the right way to get these parameters. It feels off.